src/SFA.DAS.Commitments.Application/Commands/UpdateApprenticeshipStopDate/UpdateApprenticeshipStopDateCommandValidator.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using FluentValidation;
+
+namespace SFA.DAS.Commitments.Application.Commands.UpdateApprenticeshipStopDate
+{
+    public sealed class UpdateApprenticeshipStopDateCommandValidator : AbstractValidator<UpdateApprenticeshipStopDateCommand>
+    {
+        public UpdateApprenticeshipStopDateCommandValidator()
+        {
+            RuleFor(x => x.AccountId).GreaterThan(0);
+            RuleFor(x => x.ApprenticeshipId).GreaterThan(0);
+            RuleFor(x => x.StopDate).LessThanOrEqualTo(DateTime.Today);
```

Comment:
We use the validators only to validate request completeness - that the request looks reasonable - rather than checking the specific values used are acceptable. Such business rule validation should be done in the handler itself. (Actually, it looks like this is present.)

We should not use the static DateTime class. In the handler, we have an injected/able _currentDateTime that allows testing to simulate any date (aka "time machine").

Reply:
I've removed the DateTime.Today as it is being handled in the Handler. I have added a check in the StopDate being > DateTime.MinValue as that is the default if you don't add anything to the request